.vscode-config/codespaces-launch.json

```diff
@@ -15,55 +15,56 @@
     "configurations": [
         {
             "name": "IoT Telemetry",
-            "type": "gdb",
+            "type": "cppdbg",
+            "program": "${workspaceFolder}/build/sdk/samples/iot/paho_iot_hub_telemetry_sample",
             "request": "launch",
-            "env": {
-                "AZ_IOT_HUB_DEVICE_ID": "${input:device_id}",
-                "AZ_IOT_HUB_HOSTNAME": "${input:hostname}",
-                "AZ_IOT_DEVICE_X509_CERT_PEM_FILE_PATH": "/workspaces/azure-sdk-for-c/cert/device_cert_store.pem"
-            },
-            "target": "./build/sdk/samples/iot/paho_iot_hub_telemetry_sample",
+            "environment": [
+              {"name": "AZ_IOT_HUB_DEVICE_ID","value": "${input:device_id}"},
+              {"name": "AZ_IOT_HUB_HOSTNAME","value": "${input:hostname}"},
+              {"name": "AZ_IOT_DEVICE_X509_CERT_PEM_FILE_PATH","value": "${workspaceFolder}/cert/device_cert_store.pem"},
+            ],
             "cwd": "${workspaceRoot}",
             "preLaunchTask": "build"
```

Comment:
I would make this more flexible by having at least 2 tasks.. one for rebuild everything ( like the one we already have ) and another one for just calling build to re-build w/o re-generating the project.  But that's out of this PR scope anyway :)

Reply:
agreed. hopefully will tackle in the future.